spanner_orm/tests/decorator_test.py

```diff
@@ -0,0 +1,96 @@
+from unittest import mock
+# python3
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from absl.testing import parameterized
+from spanner_orm import decorator
+
+
+# pylint: disable=redundant-keyword-arg
+# pylint: disable=no-value-for-parameter
+class DecoratorTest(parameterized.TestCase):
+
+  @parameterized.parameters(
+      (decorator.transactional_read, 'run_read_only'),
+      (decorator.transactional_write, 'run_write'),
+  )
+  @mock.patch('spanner_orm.api.SpannerApi', autospec=True)
+  def test_transactional_injects_new_transaction(self, decorator_in_test,
+                                                 api_to_mock, mock_api):
```

Comment:
is this right? the argument provided by @mock.patch is interleaved between the arguments provided by @parameterized.parameters?

Reply:
it is not interleaved, but I guess `api_to_mock` and `mock_api` sounds too similar. `mock_api` is the one coming from mock.patch. I will try to rename them better